client/state/data-layer/wpcom-http/test/http.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { WPCOM_HTTP_REQUEST } from 'state/action-types';
+
+/**
+ * Internal dependencies
+ */
+import {
+	apiVersionPattern,
+	apiNamespacePattern,
+	http
+} from '../actions';
+
+describe( 'http', () => {
+	describe( 'apiVersion shape', () => {
+		it( 'should not accept `1` as a valid version', () => {
+			expect( apiVersionPattern.test( '1' ) ).to.be.false;
+		} );
+
+		it( 'should accept `v1` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1' ) ).to.be.true;
+		} );
+
+		it( 'should not accept `v1.` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1.' ) ).to.be.false;
+		} );
+
+		it( 'should accept `v1.3` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1.3' ) ).to.be.true;
+		} );
```

Comment:
maybe we could make a list of valid versions and invalid versions and then run the test over that list. I think it would remove a lot of redundant code (which isn't actually helping us understand the functionality better) and leave us with a cleaner/simpler way of assessing what _should_ and _shouldn't_ be acceptable values.

```js
describe( 'apiVersion RegExp pattern', () => {
	const validVersions = [ 'v1', 'v1.3' ];
	const invalidVersions = [ '1', 'v1.' ];

	it( 'should pass valid versions', () => {
		validVersions.forEach( v => expect( apiVersionPattern.test( v ) ).to.be.true );
	} );

	it( 'should fail invalid versions', () => {
		invalidVersions.forEach( v => expect( apiVersionPattern.test( v ) ).to.be.false );
	} );
} );
```

now we can more easily add test cases to the tests while keeping the suite small.

> note! these are spot-check tests which catch a very limited set of inputs. we need to be aware of the limitation here